pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -1059,6 +1095,137 @@ func testHTTPsGoTLSCaptureAlreadyRunningContainer(t *testing.T, cfg *config.Conf
 	checkRequests(t, tr, expectedOccurrences, reqs)
 }
 
+// TLS classification tests
+func TestTLSClassification(t *testing.T) {
+	log.SetupLogger(seelog.Default, "debug")
+
+	cfg := testConfig()
+	cfg.ProtocolClassificationEnabled = true
+	cfg.CollectTCPConns = true
+
+	if !classificationSupported(cfg) {
+		t.Skip("TLS classification platform not supported")
+	}
+
+	// testContext shares the context of a given test.
+	// It contains common variable used by all tests, and allows extending the context dynamically by setting more
+	// attributes to the `extras` map.
+	type testContext struct {
+		// A dynamic map that allows extending the context easily between phases of the test.
+		extras map[string]interface{}
+	}
+
+	type tlsTest struct {
+		name            string
+		context         testContext
+		preTracerSetup  func(t *testing.T, ctx testContext)
+		postTracerSetup func(t *testing.T, ctx testContext)
+		validation      func(t *testing.T, ctx testContext, tr *Tracer)
+		teardown        func(t *testing.T, ctx testContext)
+	}
+	tests := []tlsTest{}
+	for _, tlsVersion := range []string{"-tls1", "-tls1_1", "-tls1_2", "-tls1_3"} {
+		tests = append(tests, tlsTest{
+			name: "TLS" + tlsVersion + "_docker",
+			context: testContext{
+				extras: make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				clientSuccess := false
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					time.Sleep(5 * time.Second)
+					clientSuccess = prototls.RunClientOpenssl(t, "localhost", "44330", tlsVersion)
+					wg.Done()
+				}()
+				prototls.RunServerOpenssl(t, "44330", "-www")
+				wg.Wait()
+				if !clientSuccess {
+					t.Fatalf("openssl client failed")
+				}
```

Comment:
it's not possible this way because I check the client and server behave _correctly_ (scope connection)
client waiting for "Verify return code" and server "exited with code"  (the server exit after 1 accept)

Reply:
the problem is that you are making a flaky test, if it takes to the server more than 5 seconds to run locally